dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/python/PythonGateway.java

```diff
@@ -171,10 +171,11 @@ public Map<String, Long> getCodeAndVersion(String projectName, String processDef
         }
 
         ProcessDefinition processDefinition = processDefinitionMapper.queryByDefineName(project.getCode(), processDefinitionName);
+        // In the case project exists, but current process definition still not created, we should also return the init version of it
         if (processDefinition == null) {
-            String msg = String.format("Can not find valid process definition by name %s", processDefinitionName);
-            logger.error(msg);
-            throw new IllegalArgumentException(msg);
+            result.put("code", CodeGenerateUtils.getInstance().genCode());
+            result.put("version", 0L);
+            return result;
```

Comment:
We need to return the init code and version here, as I said in the comment @hiSandog 
> In the case project exists, but current process definition still not created, we should also return the init version of it